graylog2-server/src/main/java/org/graylog2/indexer/MongoIndexSetRegistry.java

```diff
@@ -54,6 +55,24 @@ public MongoIndexSetRegistry(IndexSetService indexSetService,
     }
 
     @Override
+    public Optional<IndexSet> get(final String indexSetId) {
+        return findAllMongoIndexSets().stream()
+                .filter(mongoIndexSet -> mongoIndexSet.getConfig().id().equals(indexSetId))
+                .map(mongoIndexSet -> (IndexSet) mongoIndexSet)
+                .findFirst();
+    }
+
+    @Override
+    public Optional<IndexSet> getForIndexName(String indexName) {
+        for (MongoIndexSet indexSet : findAllMongoIndexSets()) {
```

Comment:
Can't we do a prefix comparison on an index here? Anyway, we can address this later as it is only a performance improvement, and I'm not even sure how often we do this.